.circleci/config.yml

```diff
@@ -845,161 +845,202 @@ jobs:
-            branches:
-              ignore:
-                - master
-                - /^test-[\d|.]+$/
-                - /^release-[\d|.]+$/
       - run-unit-test:
           requires:
             - prefetch-crates
-          filters:
-            branches:
-              ignore:
-                - master
-                - /^test-[\d|.]+$/
-                - /^release-[\d|.]+$/
       - run-crypto-unit-test:
           requires:
             - prefetch-crates
-          filters:
-            branches:
-              ignore:
-                - master
-                - /^test-[\d|.]+$/
-                - /^release-[\d|.]+$/
       - build-benchmark:
           requires:
             - prefetch-crates
-          filters:
-            branches:
-              ignore:
-                - master
-                - /^test-[\d|.]+$/
-                - /^release-[\d|.]+$/
       - build-docs:
           requires:
             - lint
+      - docker-ci-build:
+          base-image: circleci
           filters:
             branches:
-              ignore:
-                - gh-pages
-                - /^test-[\d|.]+$/
-                - /^release-[\d|.]+$/
-      - deploy-docs:
-          requires:
-            - build-docs
+              only:
+                - auto
+                - canary
+      - docker-ci-build:
+          base-image: centos
           filters:
             branches:
-              only: master
-      - check-breaking-change:
-          requires:
-            - prefetch-crates
+              only:
+                - auto
+                - canary
+      - docker-ci-build:
+          #sccache fails to execute in alpine.
+          base-image: alpine
           filters:
             branches:
-              only: master
-      - docker-pre-publish:
-          context: docker
+              only:
+                - auto
+                - canary
+      - docker-ci-build:
+          base-image: arch
           filters:
             branches:
               only:
                 - auto
-      - docker-publish:
-          context: docker
+                - canary
+
+  pull-request-workflow:
+    jobs:
+      - lint-docker:
           filters:
             branches:
-              only:
-                - /^test-[\d|.]+$/
-                - /^release-[\d|.]+$/
-      - code_coverage:
-          context: libra_ci
+              only: /^pull\/.+$/
+      - terraform:
           filters:
             branches:
-              only:
-                - /^test-[\d|.]+$/
-                - master
-      # Will build and push daily,
-      # or when relevant files change.
-      - docker-ci-build:
-          context: docker
-          base-image: circleci
+              only: /^pull\/.+$/
+      - prefetch-crates:
           filters:
             branches:
-              only:
-                - /^test-[\d|.]+$/
-                - master
+              only: /^pull\/.+$/
+      - lint:
+          requires:
+            - prefetch-crates
```

Comment:
Curious why we don't have the branch filters here.  Does that mean this job will trigger if and only if `prefetch-crates` triggers and runs?